deflate.c

```diff
@@ -813,7 +816,12 @@ int ZEXPORT deflate (strm, flush)
     /* Write the header */
     if (s->status == INIT_STATE) {
         /* zlib header */
+#if MIN_LOOKAHEAD < 256
         uInt header = (Z_DEFLATED + ((s->w_bits-8)<<4)) << 8;
+#else
+        uInt optimized_cinfo = (s->w_bits > 9) ? s->w_bits - 8 : 0;
+        uInt header = (Z_DEFLATED + (optimized_cinfo<<4)) << 8;
+#endif
```

Comment:
As far as i can understand, a 256 byte window is not supported at all #171. Therefore, it is more correct to write like this:
```diff
-        uInt header = (Z_DEFLATED + ((s->w_bits-8)<<4)) << 8;
+        uInt optimized_cinfo = (s->w_bits > 9) ? s->w_bits - 8 : 0;
+        uInt header = (Z_DEFLATED + (optimized_cinfo<<4)) << 8;

```